native-sql-engine/cpp/src/codegen/arrow_compute/ext/actions_impl.cc

```diff
@@ -868,6 +868,173 @@ class MinAction<DataType, CType, precompile::enable_if_decimal<DataType>>
   uint64_t length_ = 0;
 };
 
+/// String ///
+template <typename DataType, typename CType>
```

Comment:
can we merge this into the number/decimal type?

Reply:
I found there are some differences between them, including:

- cache_.resize(max_group_id + 1, 0) and cache_.resize(max_group_id + 1, "")
- builder_->AppendValues does not work for StringBuilder
- Arrow MinMax API does not work for string

Reply:
I found there are some differences between them, including:

- cache_.resize(max_group_id + 1, 0) and cache_.resize(max_group_id + 1, "")

- builder_->AppendValues does not work for StringBuilder

- Arrow MinMax API does not work for string